Automattic/wp-calypso
# Accessiblity: Fix Help Center portal injection

Related to https://github.com/Automattic/wp-calypso/issues/96926

## Proposed Changes

The Help Center injected a portal that it didn't use until it was opened. This portal had incorrect aria-x values until the Help Center is open. Namely `aria-labelledby` and `aria-modal=true`. Both of these values are incorrect when the Help Center is closed. 

This PR fixes this by only injecting the portal when the Help Center is open.

## Why are these changes being made?
To improve the accessibility score and thus SEO.

## Testing Instructions

1. Open and close the Help Center in Calypso my home and make sure all works well.
2. In `apps/help-center`, run `yarn dev --sync`.
3. Open the Help Center in the editor multiple times and make sure all works. 
4. Go to /setup/onboarding.
5. Run Lighthouse in DevTools and you should get an Accessibility score of 95. The 5 points are issues in the goals page, not Stepper.
